src/z.ts

```diff
@@ -44,7 +44,7 @@ const completionSpec: Fig.Spec = {
               // to work regardless of any amount over that limit.
               // Fig should defer assigning priority to z.
               // 75 added to keep args above options.
-              priority: 75 + weight,
+              priority: 75 + (weight >= 100 ? 100 : weight),
```

Comment:
This isn't really what I was thinking. My idea was more like this: the max sum of all the frecencies of z is either a custom value or 9000 https://github.com/rupa/z/blob/b82ac78a2d4457d2ca09973332638f123f065fd1/z.sh#L72
We should assume that all the items are going to be in the range `[0, MAX || 9000]`.
This should do the tweak:
```suggestion
              priority: 75 + (weight * 25 / (maxFrecency || 9000)), // where maxFrecency is read executing a shell command: `echo "$_Z_MAX_SCORE"`
```
Ping me for help!

Reply:
oooh that makes sense! how would we go about executing another shell command here?

Reply:
also, when echoing that shell variable even though I have z active, its empty 
![Uploading image.png…]().

Reply:
I'm using https://github.com/agkozak/zsh-z , which is said to be more stable and more performant

Reply:
Updated!

Reply:
Ok awesome, pushed up the commit with the suggestion!